### PR TITLE
feat(b2c): support a11y for Slider

### PR DESCRIPTION
**Pull request by Yakutoc**
Добавить поддержку доступности согласно [примеру(Horizontal Multi-Thumb Slider)](https://www.w3.org/WAI/ARIA/apg/example-index/slider/slider-multithumb.html)  

Проверить на доступность в 

**macOS**

- [VoiceOver](https://www.apple.com/ru/accessibility/vision/) - (нативные возможности системы);

**Windows** 

- [JAWS](https://www.freedomscientific.com/products/software/jaws/) (платный[ограничения в 40 мин пользования]);
- [NVDA](https://www.nvaccess.org/about-nvda/) (бесплатный);

:baby_chick: Download canary assets:

[colors--canary.218.3485919734.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.218.3485919734.xml)  
[color_sbermarket_ios-swift--canary.218.3485919734.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.218.3485919734.swift)  
[color_sbermarket_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.218.3485919734.kt)  
[color_sbermarket_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.218.3485919734.ts)  
[color_sberprime_ios-swift--canary.218.3485919734.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.218.3485919734.swift)  
[color_sberprime_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.218.3485919734.kt)  
[color_sberprime_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.218.3485919734.ts)  
[PlasmaTokensColor--canary.218.3485919734.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.218.3485919734.swift)  
[typo_mage_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.218.3485919734.kt)  
[typo_mage_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.218.3485919734.ts)  
[typo_plasma_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.218.3485919734.kt)  
[typo_plasma_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.218.3485919734.ts)  
[typo_ruler_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.218.3485919734.kt)  
[typo_ruler_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.218.3485919734.ts)  
[typo_sage_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.218.3485919734.kt)  
[typo_sage_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.218.3485919734.ts)  
[typo_sbermarket_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.218.3485919734.kt)  
[typo_sbermarket_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.218.3485919734.ts)  
[typo_soulmate_kotlin--canary.218.3485919734.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.218.3485919734.kt)  
[typo_soulmate_react-native--canary.218.3485919734.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.218.3485919734.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.113.0-canary.218.3485919734.0
  # or 
  yarn add @salutejs/plasma-b2c@1.113.0-canary.218.3485919734.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

File: packages/plasma-b2c/src/components/Slider/Double.tsx [hunk 1]
```diff
@@ -20,6 +20,14 @@ export interface SliderProps {
      * Компонент неактивен
      */
     disabled?: boolean;
+    /**
+     * Ярлык, определяющий назначение ползунка, например «Минимальная цена» [a11y].
+     */
+    ariaLabel?: string[];
+    /**
+     * Размера увеличенного шага (для клавишь PageUp, PageDown)
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Double.tsx:28]: небольшая опечатка
`клавишь => клавиш`

File: packages/plasma-b2c/src/components/Slider/Double.tsx [hunk 2]
```diff
@@ -142,6 +159,9 @@ export const Slider: React.FC<SliderProps> = ({ min, max, value, disabled, onCha
         [onChangeCommitted, value],
     );
 
+    const [ariaLabelLeft = '', ariaLabelRight = ''] = ariaLabel || [];
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Double.tsx:162]: а тут обязательно значения по умолчанию устаналивать?

File: packages/plasma-b2c/src/components/Slider/Handle.tsx [hunk 3]
```diff
@@ -27,74 +14,19 @@ const HandleStyled = styled.div`
...
-`;
-
-function getValue(handleCenterXRelative: number, stepSize: number, min: number, max: number) {
-    const newValue = Math.round(handleCenterXRelative / stepSize) + min;
+export const Handle = forwardRef<HTMLDivElement, HandleProps>(
+    (
+        { stepSize, onChangeCommitted, onChange, xPosition, min, max, bounds = [], zIndex, disabled, side, ...rest },
+        ref,
+    ) => {
+        const lastOnChangeValue = useRef<number | null>(null);
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Handle.tsx:22]: *минор*

тут по-моему не обязательно указывать в типе null

File: packages/plasma-b2c/src/components/Slider/Handle.tsx [hunk 4]
```diff
@@ -117,23 +49,39 @@ export const Handle = React.forwardRef<HTMLDivElement, HandleProps>(
         const [leftValueBound, rightValueBound] = bounds;
         const leftPositionBound = leftValueBound ? (leftValueBound - min) * stepSize : null;
         const rightPositionBound = rightValueBound ? (rightValueBound - min) * stepSize : null;
+
+        const position = typeof xPosition === 'number' ? { x: xPosition, y: 0 } : undefined;
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Handle.tsx:53]: понимаю, что так было до этого, но мб проверять просто на существование xPosition, а не его тип?

File: packages/plasma-b2c/src/components/Slider/Double.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Double.tsx:162]: @neretin-trike Ну, в принципе что с `undefined`, что с `''` поведение одинаковое. Можем и убрать.

File: packages/plasma-b2c/src/components/Slider/Single.tsx [hunk 5]
```diff
@@ -20,6 +20,14 @@ export interface SliderProps {
      * Компонент неактивен
      */
     disabled?: boolean;
+    /**
+     * Ярлык, определяющий назначение ползунка, например «Минимальная цена» [a11y].
+     */
+    ariaLabel?: string;
+    /**
+     * Размера увеличенного шага (для клавишь PageUp, PageDown).
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Single.tsx:28]: тут тоже надо поправить `клавишь => клавиш`

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Single.tsx:28]: Уже поправил )

File: packages/plasma-b2c/src/components/Slider/Handle.tsx (hunk 4 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Handle.tsx:53]: @neretin-trike Для понимания, что именно в такой проверки тебя смущает?

File: packages/plasma-b2c/src/components/Slider/Thumb.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+export const Thumb = ({
+    onChangeCommitted,
+    stepSize,
+    xPosition = 0,
+    min,
+    max,
+    value,
+    bounds,
+    ariaValueMin = min,
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Slider/Thumb.tsx:63]: расскажи пожалуста вкратце, зачем для ariaValueMin отдельный пропс заводить?

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Thumb.tsx:63]: @neretin-trike 

[prop min](url) участвует в расчетах положения указателя слайдера и шага движения, что бы не мешать эту логику с простой информацией - завел отдельный **prop**

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

File: packages/plasma-b2c/src/components/Slider/Handle.tsx (hunk 4 above)
Comment (neretin-trike) [reply packages/plasma-b2c/src/components/Slider/Handle.tsx:53]: решили оставить как есть, т.к. иначе нужно делать дополнительную проверку на 0

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Handle.tsx:53]: После обсуждения с @neretin-trike решили оставить как есть.

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-b2c/src/components/Slider/Double.tsx [hunk 7]
```diff
@@ -37,7 +45,16 @@ function getXCenterHandle(handle: HTMLDivElement) {
...
-export const Slider: React.FC<SliderProps> = ({ min, max, value, disabled, onChangeCommitted, onChange }) => {
+export const Slider: React.FC<SliderProps> = ({
+    min,
+    max,
+    value,
+    disabled,
+    onChangeCommitted,
+    onChange,
+    ariaLabel,
+    multipleStepSize,
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/Double.tsx:56]: если вот тут добавить оно прорастет же в доку?

File: packages/plasma-b2c/src/components/Slider/Double.tsx [hunk 8]
```diff
@@ -20,6 +20,14 @@ export interface SliderProps {
      * Компонент неактивен
      */
     disabled?: boolean;
+    /**
+     * Ярлык, определяющий назначение ползунка, например «Минимальная цена» [a11y].
+     */
+    ariaLabel?: string[];
+    /**
+     * Размера увеличенного шага (для клавиш PageUp, PageDown)
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/Double.tsx:28]: нужно уточнить что это проценты

File: packages/plasma-b2c/src/components/Slider/Thumb.tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            if (Home === keyCode) {
+                data.x = 0;
+            }
+
+            if (PageUp === keyCode) {
+                data.x = xPosition + computedMultipleSteps;
+                data.deltaX = computedMultipleSteps;
+            }
+
+            if (PageDown === keyCode) {
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/Thumb.tsx:116]: мне почему то хочется видеть switch тут но это вкусовщина

File: packages/plasma-b2c/src/components/Slider/utils.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+ */
+export function getOffsets(
+    ref: ((instance: HTMLDivElement | null) => void) | MutableRefObject<HTMLDivElement | null> | null,
+    side?: 'left' | 'right',
+): number[] {
+    if (!ref || !('current' in ref) || !ref.current || !side) {
+        return [0, 0];
+    }
+
+    const size = ref.current.clientWidth;
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/Slider/utils.ts:32]: а каким образом у нас clientWidth имеет значение left или right ?

File: packages/plasma-b2c/src/components/Slider/Double.tsx (hunk 8 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Double.tsx]: Добавил

File: packages/plasma-b2c/src/components/Slider/Double.tsx (hunk 7 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/Double.tsx]: не понял твоего вопроса.

File: packages/plasma-b2c/src/components/Slider/utils.ts (hunk 10 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Slider/utils.ts:32]: Это с какой стороны находится элемент ползунка на отрезке. 

Справа (1 ползунок) или слева (второй ползунок).

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-218/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-218/
  ui storybook: http://plasma.sberdevices.ru/pr-218/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-218/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-218/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-218/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**